src/core/lib/event_engine/posix_engine/tcp_socket_utils.cc

```diff
@@ -49,13 +51,23 @@
 #include <grpc/impl/codegen/grpc_types.h>
 #include <grpc/support/log.h>
 
-#include "src/core/lib/event_engine/posix_engine/tcp_socket_utils.h"
 #include "src/core/lib/gprpp/host_port.h"
 
 #ifdef GRPC_HAVE_UNIX_SOCKET
 #include <sys/un.h>
 #endif
 
+// If expr evaluates to absl Not-OK status, then execute expr2 and return the
+// evaluated status.
+#define RETURN_IF_ERROR(expr, expr2)    \
```

Comment:
The single-argument version is more general. I'd recommend keeping the absl::cleanup as you had it before, and standardizing on a general solution that works across gRPC.

Reply:
Moved it to src/core/lib/gprpp/status_helper.h file now.